client/src/main/java/io/split/client/SplitClientImpl.java

```diff
@@ -145,7 +147,31 @@ private TreatmentLabelAndChangeNumber getTreatment(String matchingKey, String bu
                 return new TreatmentLabelAndChangeNumber(parsedSplit.defaultTreatment(), "killed", parsedSplit.changeNumber());
             }
 
+            /*
+             * There are three parts to a single Split: 1) Whitelists 2) Traffic Allocation
+             * 3) Rollout. The flag inRollout is there to understand when we move into the Rollout
+             * section. This is because we need to make sure that the Traffic Allocation
+             * computation happens after the whitelist but before the rollout.
+             */
+            boolean inRollout = false;
+
             for (ParsedCondition parsedCondition : parsedSplit.parsedConditions()) {
+
+                if (!inRollout && parsedCondition.conditionType() == ConditionType.ROLLOUT) {
+
+                    if (parsedSplit.trafficAllocation() < 100) {
+                        // if the traffic allocation is 100%, no need to do anything special.
+                        int bucket = Splitter.getBucket(bucketingKey, parsedSplit.trafficAllocationSeed());
+
+                        if (bucket >= parsedSplit.trafficAllocation()) {
+                            // out of split
+                            return new TreatmentLabelAndChangeNumber(parsedSplit.defaultTreatment(), "not in split", parsedSplit.changeNumber());
```

Comment:
Can this `"not in split"` be a constant in a LABELs class or enum like other SDKs?
Same for `"killed"`, `"exception"` etc.